pkg/crdconversion/crdconversion.go

```diff
@@ -104,8 +106,28 @@ func (crdWh *crdConversionWebhook) run(stop <-chan struct{}) {
 			log.Error().Err(err).Msg("crd-converter webhook HTTP server failed to start")
 			return
 		}
+		wg.Done()
 	}()
 
+	healthMux := http.NewServeMux()
+
+	healthMux.HandleFunc(webhookHealthPath, healthHandler)
+
+	server2 := &http.Server{
+		Addr:    fmt.Sprintf(":%d", 9095),
+		Handler: healthMux,
+	}
+
+	go func() {
+		if err := server2.ListenAndServe(); err != nil {
+			log.Error().Err(err).Msg("crd-converter health server failed to start")
+			return
+		}
+		wg.Done()
+	}()
+
+	wg.Wait()
```

Comment:
Why is the Wait() required?

You also need to allow shutting down server2 similar to server1 (L135)

Reply:
Also I think, if both server fail to start, will the program wait here forever?

Reply:
This needs to be fixed as discussed offline.

Reply:
a bit confused by the wait too

Reply:
removed the wait and shut down the server